### PR TITLE
Cleanup of polling code

### PR DESCRIPTION
**Pull request by Jordie0608**
Cleans up and improves polling code by
- Removing various uneeded checks and validations
- Adding error reporting for Execute() calls
- Unspaghettifying handling of NextRow()
- Removing a level of indentation from almost the entire file
- Condensing bits of copied code
- Adding more user notification
- _Making text replies actually sanitized_

Tested to work.

### CODE REVIEW

**Comment by lzimann** [comment]
a few days late, honk

**Comment by duncathan** [comment]
this would have had little to no impact on the headmin elections

File: code/modules/mob/new_player/poll.dm [hunk 1]
```diff
@@ -3,480 +3,322 @@
...
-
-		var/adminrank = "Player"
-		if(usr && usr.client && usr.client.holder)
-			adminrank = usr.client.holder.rank
-
+	var/adminrank = poll_check_voted(pollid, "poll_textreply")
+	if(!adminrank)
+		return
+	replytext = sanitizeSQL(replytext)
+	if(!length(replytext) > 0 || !length(replytext) <= 8000)
```
Comment (YotaXP) [inline code/modules/mob/new_player/poll.dm]: Did you test this?  I don't think this works...

File: code/modules/mob/new_player/poll.dm [hunk 2]
```diff
@@ -3,480 +3,322 @@
...
+		return
+	var/i = text2num(query_get_choicelen.item[1])
+	var/DBQuery/query_hasvoted = dbcon.NewQuery("SELECT id FROM [format_table_name("poll_vote")] WHERE pollid = [pollid] AND ckey = '[ckey]'")
+	if(!query_hasvoted.Execute())
+		var/err = query_hasvoted.ErrorMsg()
+		log_game("SQL ERROR obtaining id from poll_vote table. Error : \[[err]\]\n")
+		return
+	while(i)
+		if(query_hasvoted.NextRow())
+			i--
```
Comment (YotaXP) [inline code/modules/mob/new_player/poll.dm]: Won't this be an infinite loop if there aren't enough rows?

File: code/modules/mob/new_player/poll.dm [hunk 3]
```diff
@@ -3,480 +3,322 @@
...
+	var/adminrank = "Player"
+	if(client.holder)
+		adminrank = client.holder.rank
+	var/DBQuery/query_insert = dbcon.NewQuery("INSERT INTO [format_table_name("poll_vote")] (datetime, pollid, optionid, ckey, ip, adminrank) VALUES (Now(), [pollid], [optionid], '[ckey]', '[client.address]', '[adminrank]')")
+	if(!query_insert.Execute())
+		var/err = query_insert.ErrorMsg()
+		log_game("SQL ERROR adding vote to table. Error : \[[err]\]\n")
+		return
+	usr << "<span class='notice'>Vote successful.</span>"
+	usr << browse(null,"window=playerpoll")
```
Comment (YotaXP) [inline code/modules/mob/new_player/poll.dm]: These should probably be an `if...else`.  Also, the player should be notified that the vote failed to take.

**Pull request merged**